src/server/network/network.ts

```diff
@@ -22,14 +24,46 @@ export class UDPServer {
     this.listeners = new Array<Listener>();
 
     teams.forEach((team: TeamData) => {
-      this.listeners.push({
-        server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
-        team: team,
-      });
-    });
-
-    this.listeners.forEach((listener: Listener) => {
-      this.initialiseListener(listener);
+      // Determine if we are listening on a IPv4 or IPv6 address and resolve any DNS names
+      dns.lookup(
+        team.address,
+        { family: 0, all: false },
+        (
+          error: NodeJS.ErrnoException | null,
+          address: string,
+          family: number,
+        ) => {
+          if (family === 6) {
+            const listener = {
+              server: dgram.createSocket({ type: 'udp6', reuseAddr: true }),
+              ipv6: true,
+              team: {
+                name: team.name,
+                address: address,
+                port: team.port,
+              },
+            };
+            this.initialiseListener(listener);
```

Comment:
This should just be a type with a constructor rather than an initaliseListener function. Any time you have a function that is`initialiseSomething` it's an indication that your classes are leaking into eachother